### PR TITLE
feat: Group events into live, past and upcoming

### LINKED ISSUE #152: Grouping events in the event list
To be done after #156 

Currently all events are shown in an event list.

**Expected:** Events should be grouped under live, past and future on basis of event start-end time.

### PR DESCRIPTION
**Pull request by mejariamol**
- Implements comparable interface on event model
    - sorts events in order of live, upcoming and past
    - for both past, lately ended will be before in list
    - for both live, lately started will be before in list
    - for both upcoming earliest will be before
- Implements sticky header adapter on eventslistadapter
- Modifies event layout

fixes #152

screenshot_2017-06-23-18-13-57

### CODE REVIEW

**Comment by mejariamol** [comment]
@iamareebjamal @rohanagarwal94 @Shailesh351 @Ronak-59 @mahikaw please review

**Comment by iamareebjamal** [comment]
I think showing date will look better as subtitle below title

**Comment by iamareebjamal** [comment]
@mejariamol Date functions are repeated a lot. Create a date util class and also compare util if the code for comparison is also shared

**Comment by mejariamol** [comment]
>Date functions are repeated a lot. Create a date util class

But it will include creating `DateFormat` class and setting timezone to UTC on each time it is used. With present approach `DateFormat` is created once and used at that spot.

**Comment by iamareebjamal** [comment]
You can create a static DateFormat with UTC

**Comment by iamareebjamal** [comment]
BTW, there is no need to set UTC, just show the date in Event timezone, not UTC

**Comment by mejariamol** [comment]
It is converted into UTC for comparison

**Comment by mejariamol** [comment]
@iamareebjamal Some tests are failing due to `java.text.DateFormat` which is used for parsing date. It is not thread safe.
>Date formats are not synchronized. It is recommended to create separate format instances for each thread. If multiple threads access a format concurrently, it must be synchronized externally.

from java doc. Hence it is doing job in the app as we are sorting in async but failing tests. 

shall we pass the tests for this?

**Comment by iamareebjamal** [comment]
The tests are failing due to local time being different in your settings and CI settings. We can leave the tests for now, but create an issue for the same. And that's why I asked to move the date logic out of presenter so that we can mock it in the tests

**Comment by mejariamol** [comment]
Ok. Will try using Joda-Time instead which is more clearer.

**Comment by iamareebjamal** [comment]
Nope, that library is huge and causes severe performance issues on Android! This is a 15 liner solution

Take a look at DateUtils in open android project

**Comment by mejariamol** [comment]
Actually with present code everything is working fine in the app.
please review for other things.
I am not getting why test terminate when comes at compare function where `DatetimeFormat` is initiated.
I tried on Joda-time , same behaviour , working for app but failing test.

Will create issue for the same.

**Comment by mejariamol** [comment]
you may run the code if got time. Please see if I am missing something.

**Comment by mejariamol** [comment]
@iamareebjamal what to do with this? It's working fine in the app. Only problem is with tests.

**Comment by iamareebjamal** [comment]
Will take a look at it in about an hour

**Comment by iamareebjamal** [comment]
@mejariamol Please look at the changes I made in last commit and test if everything's working correctly

**Comment by mejariamol** [comment]
@iamareebjamal yeah. working fine. Thanks :)

**Comment by mejariamol** [comment]
@arp95 @rohanagarwal94 @Shailesh351 guys, please review

**Comment by mejariamol** [comment]
@iamareebjamal please review. Tests were failing due to IEventRepository's name mismatch. Corrected.

**Comment by iamareebjamal** [comment]
They were being fixed by #233

**Comment by mejariamol** [comment]
No problem. I have just corrected IEventRepository's name. :+1:

**Comment by mejariamol** [comment]
@iamareebjamal tests from SimpleModelTest are failing . Please take a look.

**Comment by iamareebjamal** [comment]
Working fine locally. Seems internal issue with travis or roboelectric. Looking for workaround

**Comment by iamareebjamal** [comment]
@mejariamol Will be fixed by #233

File: app/src/main/java/org/fossasia/openevent/app/events/EventsListAdapter.java [hunk 1]
```diff
@@ -62,6 +73,62 @@ void showInitialEvent() {
...
+    }
+
+    @Override
+    public EventsHeaderViewHolder onCreateHeaderViewHolder(ViewGroup viewGroup) {
+        return new EventsHeaderViewHolder(EventSubheaderLayoutBinding.inflate(LayoutInflater.from(viewGroup.getContext()), viewGroup, false));
+    }
+
+    @Override
+    public void onBindHeaderViewHolder(EventsHeaderViewHolder holder, int position) {
+        final String LIVE = "LIVE";
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/events/EventsListAdapter.java]: Use the constants declared above

File: app/src/main/java/org/fossasia/openevent/app/events/EventsListAdapter.java [hunk 2]
```diff
@@ -62,6 +73,62 @@ void showInitialEvent() {
...
+                    holder.bindHeader(PAST);
+                }
+            } else {
+                holder.bindHeader(UPCOMING);
+            }
+
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/events/EventsListAdapter.java]: This block is repeated. Create a function

**Comment by mejariamol** [comment]
@iamareebjamal done.

**Pull request merged**